### PR TITLE
Remove zerocopy and replace with bytemuck.

### LINKED ISSUE #146: Alternative to ZeroCopy
Unfortunately, ZeroCopy take a really long time to build. It's a part of Fuchsia code base, and communicating with developers or filing issues seems... problematic. Perhaps, we could make a simpler version of it for ourselves, or find an existing alternative.

### PR DESCRIPTION
**Pull request by StarArawn**
fixes #146 

I've removed `zerocopy` from the examples and replaced it with `bytemuck`. I ran all of the examples/tests and everything ran great in vulkan on my windows box.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -55,7 +55,7 @@ log = "0.4"
 png = "0.15"
 winit = "0.22"
 rand = "0.7.2"
-zerocopy = "0.3"
+bytemuck = "1.2.0"
```
Comment (kvark) [inline Cargo.toml:58]: does it have to be "1.2.0" specifically, or could be just "1"?

File: examples/cube/main.rs [hunk 2]
```diff
@@ -123,10 +126,10 @@ impl framework::Example for Example {
         let (vertex_data, index_data) = create_vertices();
 
         let vertex_buf =
-            device.create_buffer_with_data(vertex_data.as_bytes(), wgpu::BufferUsage::VERTEX);
+            device.create_buffer_with_data(bytemuck::try_cast_slice(&vertex_data).unwrap(), wgpu::BufferUsage::VERTEX);
```
Comment (kvark) [inline examples/cube/main.rs:129]: why not just `cast_slice` if we are unwrapping anyway?

File: examples/boids/main.rs [hunk 3]
```diff
@@ -204,7 +203,7 @@ impl framework::Example for Example {
         let mut particle_bind_groups = Vec::<wgpu::BindGroup>::new();
         for _i in 0..2 {
             particle_buffers.push(device.create_buffer_with_data(
-                initial_particle_data.as_bytes(),
+                bytemuck::try_cast_slice(&initial_particle_data).unwrap(),
```
Comment (kvark) [inline examples/boids/main.rs:206]: similarly in all cases: if we are just unwrapping, no need to `try_`

File: Cargo.toml (hunk 1 above)
Comment (kvark) [reply Cargo.toml:58]: this is unresolved?

**Comment by StarArawn** [comment]
@kvark 
Sorry the file didn't save and I thought it did! 😓

File: Cargo.toml [hunk 4]
```diff
@@ -55,7 +55,7 @@ log = "0.4"
 png = "0.15"
 winit = "0.22"
 rand = "0.7.2"
-zerocopy = "0.3"
+bytemuck = "1.0.0"
```
Comment (kvark) [inline Cargo.toml:58]: just "1" is good :)

**Comment by kvark** [comment]
Whoa, github actions are so fast!

**Comment by rukai** [comment]
Unfortunate that the examples now need unsafe...

**Comment by kvark** [comment]
@rukai true. Is this something you wanted to avoid? Technically, there could be a crate that implements `Pod` and `Zeroable` by reflecting the struct contents and ensuring this is safe. The benefit of going with this PR is that this crate would not need to be forced upon the users, and they don't have to pay for the compile times (which zerocopy was really bad at).

**Comment by StarArawn** [comment]
@rukai @kvark 
There is an issue open for deriving instead of implementing, and you'd need a separate crate for it to work:
https://github.com/Lokathor/bytemuck/issues/4

**Pull request merged**